crates/block-producer/src/utils.rs

```diff
@@ -12,3 +15,14 @@ pub async fn dump_transaction<P: AsRef<Path>>(dir: P, rpc_client: &RPCClient, tx
         );
     }
 }
+
+/// Convert global_state.last_finalized_timepoint to the form fo Since.
+pub fn global_state_last_finalized_timepoint_to_since(global_state: &GlobalState) -> u64 {
+    match Timepoint::from_full_value(global_state.last_finalized_timepoint().unpack()) {
+        Timepoint::BlockNumber(_) => 0,
+        Timepoint::Timestamp(time_ms) => {
+            // ATTENTION: I am not sure if I am do right. Please review intensively.
+            Since::new_timestamp_seconds(time_ms.saturating_div(1000).saturating_sub(1)).as_u64()
```

Comment:
```suggestion
            Since::new_timestamp_seconds(time_ms / 1000).as_u64()
```

Reply:
I am not sure whether we should minus 1 or not? Better to comment some explanation.

Reply:
I see. There is no necessity to minus 1. The timestamp is read from `GlobalState`. Then it must be available on-chain once. So we can assume to use the timestamp as `since` should work on-chain.